erc721-lockable/contracts/base/ERC721Lockable.sol

```diff
@@ -0,0 +1,453 @@
+     *
+     * - `from` cannot be the zero address.
+     * - `tokenId` token must be owned by `from`.
+     * - the caller must be the operator who locks the token by {lockFrom}
+     *
+     * Emits a {Unlocked} event.
+     * @param from (address) Unlock from address
+     * @param tokenId (uint256) Token ID
+     */
+    function unlockFrom(address from, uint256 tokenId) public virtual override {
+        require(
+            lockerOf(tokenId) == _msgSender(),
+            "ERC721Lockable: unlock caller is not lock operator"
+        );
+        require(
+            ERC721.ownerOf(tokenId) == from,
+            "ERC721Lockable: unlock from incorrect owner"
+        );
+
+        _beforeTokenLock(_msgSender(), from, tokenId, 0);
+
+        delete lockedTokens[tokenId];
+
+        emit Unlocked(_msgSender(), from, tokenId);
+
+        _afterTokenLock(_msgSender(), from, tokenId, 0);
+    }
+
+    /**
+     * @dev Locks `tokenId` from `from`  until `expired`.
+     *
+     * Requirements:
+     *
+     * - `tokenId` token must be owned by `from`.
+     *
+     * Emits a {Locked} event.
+     * @param operator (address) Operator account address
+     * @param from (address) Lock from address
+     * @param tokenId (uint256) Token ID
+     * @param expired (uint256) Expire timestamp
+     */
+    function _lock(
+        address operator,
+        address from,
+        uint256 tokenId,
+        uint256 expired
+    ) internal virtual {
+        require(
+            ERC721.ownerOf(tokenId) == from,
+            "ERC721Lockable: lock from incorrect owner"
+        );
+
+        _beforeTokenLock(operator, from, tokenId, expired);
+
+        lockedTokens[tokenId] = expired;
+        _lockApprovals[tokenId] = _msgSender();
+
+        emit Locked(operator, from, tokenId, expired);
+
+        _afterTokenLock(operator, from, tokenId, expired);
+    }
+
+    /**
+     * @dev Safely mints `tokenId` and transfers it to `to`, but the `tokenId` is locked and cannot be transferred.
+     *
+     * Requirements:
+     *
+     * - `tokenId` must not exist.
+     *
+     * Emits {Locked} and {Transfer} event.
+     * @param to (address) Safe lock mint to address
+     * @param tokenId (uint256) Token ID
+     * @param expired (uint256) Expire timestamp
+     * @param _data (bytes memory) Bytes data
+     */
+    function _safeLockMint(
+        address to,
+        uint256 tokenId,
+        uint256 expired,
+        bytes memory _data
+    ) internal virtual {
+        require(
+            expired > block.timestamp,
+            "ERC721Lockable: lock mint for invalid lock block time"
+        );
+
+        _safeMint(to, tokenId, _data);
+
+        _lock(address(0), to, tokenId, expired);
+    }
+
+    /**
+     * @dev See {ERC721-_burn}. This override additionally clears the lock approvals for the token.
+     */
+    function _burn(uint256 tokenId) internal virtual override {
+        address owner = ERC721.ownerOf(tokenId);
+        super._burn(tokenId);
```

Comment:
This will check if this `tokenId` was already locked before burning, correct?

Reply:
Yes, correct @deep-quality-dev